modulepath-tests/src/test/java/com/hazelcast/test/modulepath/SmokeModulePathTest.java

```diff
@@ -0,0 +1,111 @@
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.junit.After;
+import org.junit.Test;
+
+import com.hazelcast.client.HazelcastClient;
+import com.hazelcast.config.Config;
+import com.hazelcast.config.NetworkConfig;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.Member;
+import com.hazelcast.instance.HazelcastInstanceFactory;
+
+/**
+ * Basic test which checks if correct Hazelcast modules are on the modulepath. It also checks that Hazelcast members and clients
+ * are able to start and form a cluster.
+ */
+public class SmokeModulePathTest {
+
+    @After
+    public void after() {
+        HazelcastClient.shutdownAll();
+        HazelcastInstanceFactory.terminateAll();
+    }
+
+    /**
+     * Verify the Hazelcast JARs are on the modulepath.
+     */
+    @Test
+    public void testModulePath() {
+        String modulePath = System.getProperty("jdk.module.path");
+        assertNotNull("Module path was expected", modulePath);
+        assertTrue("Module path should contain hazelcast JAR",
+                modulePath.matches(".*hazelcast-[1-9][\\p{Alnum}\\-_\\.]+\\.jar.*"));
+        assertTrue("Module path should contain hazelcast-client JAR",
+                modulePath.matches(".*hazelcast-client-[\\p{Alnum}\\-_\\.]+\\.jar.*"));
+    }
+
+    /**
+     * Verify the Hazelcast modules with correct names are used.
+     */
+    @Test
+    public void testModuleNames() {
+        Set<String> hazelcastModuleNames = ModuleLayer.boot().modules().stream().map(Module::getName)
+                .filter(s -> s.contains("hazelcast")).collect(Collectors.toSet());
+        assertThat(hazelcastModuleNames, hasItems("com.hazelcast.core", "com.hazelcast.client", "com.hazelcast.tests"));
+    }
+
+    /**
+     * Verify Hazelcast members are able to start and form cluster. It also verifies the client is able to join and work with
+     * the cluster.
+     */
+    @Test
+    public void testCluster() {
+        Config config = new Config();
+
+        NetworkConfig networkConfig = config.getNetworkConfig();
+        networkConfig.getInterfaces().addInterface("127.0.0.1");
+        networkConfig.getJoin().getMulticastConfig().setEnabled(false);
+        networkConfig.getJoin().getTcpIpConfig().setEnabled(true).addMember("127.0.0.1");
+
+        HazelcastInstance hz1 = Hazelcast.newHazelcastInstance(config);
+        HazelcastInstance hz2 = Hazelcast.newHazelcastInstance(config);
+        hz1.getMap("test").put("a", "b");
+        assertClusterSize(2, hz1, hz2);
+        HazelcastInstance client = HazelcastClient.newHazelcastClient();
+        assertEquals("b", client.getMap("test").get("a"));
+    }
+
+    public static void assertClusterSize(int expectedSize, HazelcastInstance... instances) {
```

Comment:
Just out of curiosity - what is the reason why we can't use the HazelcastTestSupport.assertClusterSize? Because of the separation, we cannot use any code from the other modules?

Reply:
The test classes from other Hazelcast Maven modules are not provided as a dependency as they are not modularized. We can't simply modularize the other tests, because they share the packages with core and clients, so the compilation of this new Maven module would fail.

Reply:
Thanks for a clear explanation :) Approving.